synapse/lib/stormsvc.py

```diff
@@ -120,59 +120,63 @@ async def __anit__(self, core, sdef):
 
         self.onfini(self.proxy.fini)
 
-    async def _onTeleLink(self, proxy):
+    async def _runSvcInit(self):
 
-        clss = proxy._getClasses()
+        try:
+            await self.core._delStormSvcPkgs(self.iden)
 
-        names = [c.rsplit('.', 1)[-1] for c in clss]
+        except asyncio.CancelledError:  # pragma: no cover
+            raise
 
-        if 'StormSvc' in names:
-            self.info = await proxy.getStormSvcInfo()
+        except Exception:
+            logger.exception(f'_delStormSvcPkgs failed for service {self.name} ({self.iden})')
+
+        # Register new packages
+        for pdef in self.info.get('pkgs', ()):
 
             try:
-                await self.core._delStormSvcPkgs(self.iden)
+                # push the svciden in the package metadata for later reference.
+                pdef['svciden'] = self.iden
+                await self.core._hndladdStormPkg(pdef)
 
             except asyncio.CancelledError:  # pragma: no cover
                 raise
 
             except Exception:
-                logger.exception(f'_delStormSvcPkgs failed for service {self.name} ({self.iden})')
+                name = pdef.get('name')
+                logger.exception(f'addStormPkg ({name}) failed for service {self.name} ({self.iden})')
 
-            # Register new packages
-            for pdef in self.info.get('pkgs', ()):
+        # Set events and fire as needed
+        evts = self.info.get('evts')
+        try:
+            if evts is not None:
+                self.sdef = await self.core.setStormSvcEvents(self.iden, evts)
 
-                try:
-                    # push the svciden in the package metadata for later reference.
-                    pdef['svciden'] = self.iden
-                    await self.core.addStormPkg(pdef)
+        except asyncio.CancelledError:  # pragma: no cover
+            raise
 
-                except asyncio.CancelledError:  # pragma: no cover
-                    raise
+        except Exception:
+            logger.exception(f'setStormSvcEvents failed for service {self.name} ({self.iden})')
 
-                except Exception:
-                    name = pdef.get('name')
-                    logger.exception(f'addStormPkg ({name}) failed for service {self.name} ({self.iden})')
+        try:
+            if self.core.isleader:
```

Comment:
We cannot check self.core.nexsroot for leadership since we may be in a spawncore which has no nesxsroot.